### PR TITLE
feat(bullet): bullet does not hurt friendly entities

### PR DESCRIPTION
**Pull request by rand0m-cloud**

### CODE REVIEW

File: src/player.rs [hunk 1]
```diff
@@ -91,6 +91,7 @@ fn player_shoot(
             .insert(Bullet {
                 speed: 0.2,
                 direction: target_dir,
+                origin_team: ChickenOrDog::Chicken,
```
Comment (mwbryant) [inline src/player.rs]: I would prefer to add chicken or dog as a component over encapsulating it in the bullet

File: src/bullet.rs [hunk 2]
```diff
@@ -11,13 +9,33 @@ impl Plugin for BulletPlugin {
...
+    bullets: Query<(&Collisions, &Bullet)>,
+) {
+    // have to collect the iter to drop the entities query
+    #[allow(clippy::needless_collect)]
+    let entities_to_damage = bullets
+        .iter()
+        .flat_map(|(collisions, bullet)| {
+            collisions
+                .entities()
+                .map(move |collision| (collision, bullet))
```
Comment (mwbryant) [inline src/bullet.rs]: the logic here feels overly complex for the task being done

File: src/prelude.rs [hunk 3]
```diff
@@ -91,28 +92,3 @@ impl Default for Spawner {
...
     }
 }
-
-/// Checks if a collision event contains a bullet. If so, return the entities with the bullet as the first entity
-pub fn is_bullet_collision(event: &CollisionEvent) -> Option<(Entity, Entity)> {
-    is_layer_collision(event, Layer::Bullet)
-}
-
-/// Checks if a collision event contains the specific physics layer. If so, return the entities with the chosen layer as the first entity
-pub fn is_layer_collision(event: &CollisionEvent, layer: Layer) -> Option<(Entity, Entity)> {
```
Comment (mwbryant) [inline src/prelude.rs:101]: I think this helper function may still be useful in the future, I'd prefer to tag as dead code over deleting and having to dig it back up

File: src/bullet.rs [hunk 4]
```diff
@@ -11,40 +9,45 @@ impl Plugin for BulletPlugin {
...
+        })
+        .filter_map(|(entity, bullet)| {
+            let (_, entity_team) = entities.get(entity).ok()?;
+            if bullet.origin_team != *entity_team {
+                Some(entity)
+            } else {
+                None
+            }
+        })
```
Comment (mwbryant) [inline src/bullet.rs]: hmm thats strange, can we not clone the entity or something

File: src/prelude.rs [hunk 5]
```diff
@@ -91,28 +92,3 @@ impl Default for Spawner {
...
-
-    match [layers.0, layers.1]
-        .into_iter()
-        .position(|l| l.contains_group(layer))
-    {
-        Some(0) => Some(entities),
-        Some(1) => Some((entities.1, entities.0)),
-        _ => None,
-    }
-}
```
Comment (mwbryant) [inline src/prelude.rs]: I think this helper function may still be useful in the future, I'd prefer to tag as dead code over deleting and having to dig it back up

File: src/bullet.rs (hunk 4 above)
Comment (rand0m-cloud) [reply src/bullet.rs]: no and actually `Entity` is returned from the iter. We have to close out the reading of entities before we can mutate any.

File: src/bullet.rs (hunk 2 above)
Comment (rand0m-cloud) [reply src/bullet.rs]: the logic here is the check for bullet's origin team and the entities team. the flat map gets us to a list of entities and the bullet that collided.

File: src/bullet.rs (hunk 4 above)
Comment (mwbryant) [reply src/bullet.rs]: pls don't resolve comments chains that I start until I acknowledge the response.  I added a commit fixing this problem

File: src/bullet.rs (hunk 2 above)
Comment (mwbryant) [reply src/bullet.rs]: I see that, but differences in variable names of just the character s are often confusing,  it's fine here

**Pull request merged**